R/swagger.R

```diff
@@ -191,5 +346,58 @@ removeNaOrNulls <- function(x) {
   ret <- lapply(x, removeNaOrNulls)
   class(ret) <- class(x)
 
+  # Put example back in
+  if (saveExample) {
+    ret$example <- savedExample
+  }
+
   ret
 }
+
+#' For openapi definition
+#' @noRd
+priorizeProperty <- function(...) {
+  l <- list(...)
+  isnullordefault <- sapply(l, function(x) {isNaOrNull(x) || isTRUE(attributes(x)$default)})
+  l[[which.min(isnullordefault)]]
+}
+
+#' Check if x is JSON serializable
+#' @noRd
+isJSONserializable <- function(x) {
+  testJSONserializable <- TRUE
+  tryCatch(toJSON(x),
+           error = function(cond) {
+             # Do we need to test for specific errors?
+             testJSONserializable <<- FALSE}
+  )
+  testJSONserializable
+}
+
+#' Extract metadata on args of plumberExpression
+#' @noRd
+getArgsMetadata <- function(plumberExpression){
+  #return same format as getTypedParams or params?
+  args <- formals(eval(plumberExpression))
+  lapply(args[!names(args) %in% c("...", "res", "req")], function(arg) {
+    required <- identical(arg, formals(function(x){})$x)
```

Comment:
```suggestion
  args <- formals(eval(plumberExpression))
  required_arg <- formals(function(x){})$x
  lapply(args[!names(args) %in% c("...", "res", "req")], function(arg) {
    required <- identical(arg, required_arg)
```

Reply:
Reverted missing does not seems to be storable in `required_arg`, leaving it as it is.